python/lsst/meas/extensions/gaap/gaap.py

```diff
@@ -0,0 +1,330 @@
+# This file is part of meas_extensions_gaap
+#
+# Developed for the LSST Data Management System.
+# This product includes software developed by the LSST Project
+# (http://www.lsst.org/).
+# See the COPYRIGHT file at the top-level directory of this distribution
+# for details of code ownership.
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the LSST License Statement and
+# the GNU General Public License along with this program.  If not,
+# see <http://www.lsstcorp.org/LegalNotices/>.
+
+from __future__ import annotations
+
+__all__ = ("GaapFluxPlugin", "GaapFluxConfig", "ForcedGaapFluxPlugin", "ForcedGaapFluxConfig")
+
+from typing import Optional
+import itertools
+import lsst.afw.image as afwImage
+import lsst.afw.detection as afwDetection
+import lsst.afw.geom as afwGeom
+import lsst.meas.base
+import lsst.pex.config as pexConfig
+from lsst.ip.diffim import ModelPsfMatchTask
+from lsst.meas.base import GenericPlugin, SdssShapeAlgorithm
+from lsst.meas.base import FlagDefinitionList, FlagHandler
+from lsst.pex.exceptions import RuntimeError as pexRuntimeError
+
+PLUGIN_NAME = "ext_gaap_GaapFlux"
+
+
+class GaapConvolutionError(pexRuntimeError):
+    """Collection of any unexpected errors in GAaP during PSF Gaussianization.
+
+    The PSF Gaussianization procedure using `modelPsfMatchTask` may throw
+    exceptions for certain target PSFs. Such errors are caught until all
+    measurements are at least attempted. The complete traceback information
+    is lost, but unique error messages are preserved.
+
+    Parameters
+    ----------
+    errors : `dict` [`str`, `Exception`]
+        The values are exceptions raised, while the keys are the loop variables
+        (in `str` format) where the exceptions were raised.
+    """
+    def __init__(self, errors: dict[str, Exception], *args, **kwds):
+        message = "Problematic scaling factors = "
+        message += ", ".join(errors)
+        message += " Errors: "
+        message += " | ".join(set(msg.__repr__() for msg in errors.values()))  # msg.cpp.what() misses type
```

Comment:
I could have sworn we had something in the python standards that said that comments should be on their own line, but I can't find it now.